sklearn/manifold/spectral_embedding_.py

```diff
@@ -10,13 +10,14 @@
 import numpy as np
 from scipy import sparse
 from scipy.linalg import eigh
-from scipy.sparse.linalg import eigsh, lobpcg
+from scipy.sparse.linalg import eigsh
 from scipy.sparse.csgraph import connected_components
 from scipy.sparse.csgraph import laplacian as csgraph_laplacian
 
 from ..base import BaseEstimator
 from ..utils import check_random_state, check_array, check_symmetric
 from ..utils.extmath import _deterministic_vector_sign_flip
+from ..utils.fixes import lobpcg
```

Comment:
wouldn't it be better to check the scipy version, or try to import from it, and only import our vendored version if it fails?

Reply:
sorry, ignore this, thought you we directly importing it.